devicetypes/axis/axis-gear-st.src/axis-gear-st.groovy

```diff
@@ -0,0 +1,408 @@
+		}
+	}
+}
+
+//Send Command through setLevel()
+def on() {
+	log.info "on()"
+	sendEvent(name: "windowShade", value: "opening")
+	sendEvent(name: "switch", value: "on")
+	
+	if (state.currentVersion >= MIN_WINDOW_COVERING_VERSION) {
+		zigbee.command(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_CMD_OPEN)
+	}
+	else {
+		setLevel(100)
+	}
+}
+
+//Send Command through setLevel()
+def off() {
+	log.info "off()"
+	sendEvent(name: "windowShade", value: "closing")
+	sendEvent(name: "switch", value: "off")
+	close()
+	//zigbee.off()
+}
+
+//Command to set the blind position (%) and log the event
+def setLevel(value, rate=null) {
+	log.info "setLevel ($value)"
+	
+	Integer currentLevel = state.level
+	
+	def i = value as Integer
+	sendEvent(name:"level", value: value, displayed:true)
+	
+	if ( i == 0) {
+		sendEvent(name: "switch", value: "off")
+	}
+	else {
+		sendEvent(name: "switch", value: "on")
+	}
+	
+	if (i > currentLevel) {
+		sendEvent(name: "windowShade", value: "opening")
+	}
+    else if (i < currentLevel) {
+		sendEvent(name: "windowShade", value: "closing")
+	}
+	//setWindowShade(i)
+	
+	if (state.currentVersion >= MIN_WINDOW_COVERING_VERSION){
+		zigbee.command(CLUSTER_WINDOWCOVERING,WINDOWCOVERING_CMD_GOTOLIFTPERCENTAGE, zigbee.convertToHexString(100-i,2))
+	}
+	else {
+		zigbee.setLevel(i)
+	}
+}
+
+//Send Command through setLevel()
+def open() {
+	log.info "open()"
+	sendEvent(name: "windowShade", value: "opening")
+	if (state.currentVersion >= MIN_WINDOW_COVERING_VERSION){
+		zigbee.command(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_CMD_OPEN)
+	}
+	else {
+		setLevel(100)
+	}     
+}
+//Send Command through setLevel()
+def close() {
+	log.info "close()"
+	sendEvent(name: "windowShade", value: "closing")
+	if (state.currentVersion >= MIN_WINDOW_COVERING_VERSION){
+		zigbee.command(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_CMD_CLOSE)
+	}
+	else {
+		setLevel(0)
+	}
+}
+
+def presetPosition() {
+	log.info "presetPosition()"
+	setLevel(preset ?: state.preset ?: 50)
+}
+
+//Reporting of Battery & position levels
+def ping(){
+	log.debug "Ping() "
+	return refresh()
+}
+
+//Set blind State based on position (which shows appropriate image) 
+def setWindowShade(value) {
+	if ((value>0)&&(value<99)){
+		sendEvent(name: "windowShade", value: "partially open", displayed:true)
+	}
+	else if (value >= 99) {
+		sendEvent(name: "windowShade", value: "open", displayed:true)
+	}
+	else {
+		sendEvent(name: "windowShade", value: "closed", displayed:true)
+	}
+}
+
+//Refresh command
+def refresh() {
+	log.debug "parse() refresh"
+	def cmds_refresh = null
+    
+	if (state.currentVersion >= MIN_WINDOW_COVERING_VERSION){
+		cmds_refresh = zigbee.readAttribute(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_ATTR_LIFTPERCENTAGE) +
+							zigbee.readAttribute(CLUSTER_POWER, POWER_ATTR_BATTERY) +
+							zigbee.readAttribute(CLUSTER_BASIC, BASIC_ATTR_SWBUILDID)
+    }
+    else {
+		cmds_refresh = zigbee.readAttribute(CLUSTER_LEVEL, LEVEL_ATTR_LEVEL) +
```

Comment:
Only the first command is different here. You could condense this.

Reply:
change made to condense